docs/query.md

```diff
@@ -0,0 +1,265 @@
+# Zoekt Query Language Guide
+
+This guide explains the Zoekt query language, used for searching text within Git repositories. Zoekt queries allow combining multiple filters and expressions using logical operators, negations, and grouping. Here's how to craft queries effectively.
+
+---
+
+## Syntax Overview
+
+A query is made up of expressions. An **expression** can be:
+- A negation (e.g., `-`),
+- A field (e.g., `repo:`).
+- A grouping (e.g., parentheses `()`),
+
+Logical `OR` operations combine multiple expressions. The **`AND` operator is implicit**, meaning multiple expressions written together will be automatically treated as `AND`.
+
+---
+
+## Query Components
+
+### 1. **Fields**
+
+Fields restrict your query to specific criteria. Here's a list of fields and their usage:
+
+| Field        | Aliases | Values                 | Description                                                | Examples                               |
+|--------------|---------|------------------------|------------------------------------------------------------|----------------------------------------|
+| `archived:`  | `a:`    | `yes` or `no`          | Filters archived repositories.                             | `archived:yes`                         |
+| `case:`      | `c:`    | `yes`, `no`, or `auto` | Matches case-sensitive or insensitive text.                | `case:yes content:"Foo"`               |
+| `content:`   | `c:`    | Text (string or regex) | Searches content of files.                                 | `content:"search term"`                |
+| `file:`      | `f:`    | Text (string or regex) | Searches file names.                                       | `file:"main.go"`                       |
+| `fork:`      | `f:`    | `yes` or `no`          | Filters forked repositories.                               | `fork:no`                              |
+| `lang:`      | `l:`    | Text                   | Filters by programming language.                           | `lang:python`                          |
+| `public:`    |         | `yes` or `no`          | Filters public repositories.                               | `public:yes`                           |
+| `regex:`     |         | Regex pattern          | Matches content using a regular expression.                | `regex:/foo.*bar/`                     |
```

Comment:
This is another zoekt prefix that is unfamiliar and is a bit of a oddball: if everything is evaluated as a regular expression anyways, not sure when this would be useful.

For example:
- [with regex:](https://demo.sourcebot.dev/search?query=regex%3Afoo.*bar)
- [without regex:](https://demo.sourcebot.dev/search?query=foo.*bar)

the results are the same